src/app/shared/pipes/attribute-to-snipcart-list.pipe.ts

```diff
@@ -0,0 +1,14 @@
+import { Pipe, PipeTransform } from "@angular/core";
+
+import { ProductAttribute } from "@shared/models/classes/product-attribute.class";
+
+@Pipe({
+    name: "attributeToSnipcartList"
+})
+export class AttributeToSnipcartListPipe implements PipeTransform {
+    transform(attribute: ProductAttribute): string {
```

Comment:
the pipe only uses the nested options array of the passed attribute. Instead, you should pass the options array directly.

Also, reduce tends to reduce readability so should be used as a last resort.

This pipe can be cleaned by using map(opt) => `${opt.optionName}[${ o.priceModifier }]`

and then returning array.join("|")